apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/features/InterfaceApi.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.openstack.nova.v2_0.features;
+
+import org.jclouds.openstack.keystone.v2_0.filters.AuthenticateRequest;
+import org.jclouds.openstack.nova.v2_0.domain.InterfaceAttachment;
+import org.jclouds.openstack.nova.v2_0.options.AttachInterfaceOptions;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.SelectJson;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * interface api for nova
+ */
+@RequestFilters(AuthenticateRequest.class)
+@Consumes(MediaType.APPLICATION_JSON)
+@Path("/servers")
+public interface InterfaceApi {
+
+   @POST
+   @Consumes(MediaType.APPLICATION_JSON)
+   @SelectJson("interfaceAttachment")
+   @Path("/{server_id}/os-interface")
+   @RequestFilters(AuthenticateRequest.class)
+   InterfaceAttachment attachInterface(@PathParam("server_id") String serverId, AttachInterfaceOptions... options);
```

Comment:
Add the corresponding `@Named` annotation to this method.

Don't use a varargs parameter to express that a parameter is Optional. Overload the method to have two: one without the parameter and one with the parameter (since you don't want more than one options object being passed).